maint_tools/check_docstrings.py

```diff
@@ -0,0 +1,152 @@
+"""Utility to check dostrings.
+
+- checks donstrings of functions, classes and methods
+- checks for:
+    - find missing :obj:`` in doc string type
+    - if a function of class definition uses the fill_doc decorator properly
+"""
+
+from __future__ import annotations
+
+import ast
+
+from numpydoc.docscrape import NumpyDocString
+from rich import print
+from utils import list_classes, list_functions, list_modules
+
+# List of values to check for missing :obj:`` link
+VALUES = [
+    "integers",
+    "integer",
+    "Integer",
+    "strings",
+    "string",
+    "String",
+    "boolean",
+    "Boolean",
+    "list",
+    "List",
+    "tuple",
+    "Tuple",
+    "dict",
+    "Dict",
+    # "int",
+    "Int",
+    # "float",
```

Comment:
To be removed?

Reply:
na, this list is meant to be edited when one of us is looking for specific object that have not been documented properly: acts a bit as a 'filter'

Reply:
I see. So there would be situations where some of them don't need to tagged with :obj:

Reply:
there are a few (and also because my script catches false positive)